internal/strdist/strdist.go

```diff
@@ -125,3 +134,37 @@ func globCost(ar, br rune) Cost {
 	}
 	return Cost{SwapAB: 1, DeleteA: 1, InsertB: 1}
 }
+
+// wildcardPrefixMatch compares whether the prefixes of a and b are equal up
+// to the shortest one. The prefix is defined as the longest substring that
+// starts at index 0 and does not contain a wildcard.
+func wildcardPrefixMatch(a, b string) bool {
+	ai := strings.IndexAny(a, "*?")
+	bi := strings.IndexAny(b, "*?")
+	if ai == -1 {
+		ai = len(a)
+	}
+	if bi == -1 {
+		bi = len(b)
+	}
+	mini := min(ai, bi)
+	return a[:mini] == b[:mini]
+}
+
+// wildcardSuffixMatch compares whether the suffixes of a and b are equal up
+// to the shortest one. The suffix is defined as the longest substring that ends
+// at the string length and does not contain a wildcard.
+func wildcardSuffixMatch(a, b string) bool {
+	ai := strings.LastIndexAny(a, "*?")
+	la := 0
+	if ai != -1 {
+		la = len(a) - ai - 1
+	}
+	lb := 0
+	bi := strings.LastIndexAny(b, "*?")
+	if bi != -1 {
+		lb = len(b) - bi - 1
+	}
+	minl := min(la, lb)
```

Comment:
The opposite would be max here. At first I thought this was wrong, again because I had the prefix logic in mind and didn't realize here it's the minimum of the delta, rather than the minimum of the position, per logic above.